k2/csrc/eval.h

```diff
@@ -116,11 +118,12 @@ template <typename ContextPtrType,  // Context*  or ContextPtr ==
                                     // std::shared_ptr<Context>
           typename LambdaT>
 void Eval(ContextPtrType c, int32_t n, LambdaT &lambda) {
+  DeviceGuard guard(c);
   Eval(c->GetCudaStream(), n, lambda);
 }
 
 template <typename LambdaT>
-void EvalDevice(cudaStream_t stream, int32_t n, LambdaT &lambda) {
+static void EvalDevice(cudaStream_t stream, int32_t n, LambdaT &lambda) {
```

Comment:
Why the `static`?

Reply:
1. To mark this function being used only inside this file.
2. To reduce the number of global symbols, which I believe can reduce the link time.

Reply:
I have reverted the changes to `eval.h`.

The device guard is put only in the python wrapper.